_episodes/06-debugging.md

```diff
@@ -0,0 +1,309 @@
+>39            preprocessor: prep_timeseries
+>40        scripts:
+>41          timeseries_diag:
+>42            script: ocean/diagnostic_timeseries.py
+>~~~
+{: .solution}
+
+We change the ``projects`` value ``ukesm`` to ``tutorial``:
+~~~YAML
+19      projects:
+20        - tutorial
+~~~
+
+Also, let's add the preprocessor ``extract_region`` to the section ``prep_timeseries``:
+
+~~~YAML
+25    preprocessors:
+26      prep_timeseries: # For 0D fields
+27        annual_statistics:
+28          operator: mean
+29        extract_region:
+30          start_longitude: -10
+31          end_longitude: 40
+32          start_latitude: 27
+33          end_latitude: 70
+~~~
+
+Then, we save the file and run the recipe:
+~~~bash
+  esmvaltool -c user-config.yml recipe_example.yml
+~~~
+
+~~~
+ValueError: Tag 'tutorial' does not exist in section 'projects' of esmvaltool/config-references.yml
+2020-06-29 18:09:56,641 UTC [46055] INFO    If you suspect this is a bug or need help,
+please open an issue on https://github.com/ESMValGroup/ESMValTool/issues and
+attach the run/recipe_*.yml and run/main_log_debug.txt files from the output directory.
+~~~
+{: .error}
+
+The values for the keys ``author``, ``maintainer``, ``projects`` and ``references`` in the recipe should be known by ESMValTool.
+
+> ## ESMValTool pre-processors
+>
+> The [ESMValTool pre-processors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/recipe/preprocessor.html?highlight=preprocessor) cover a broad range of operations on the input data,
+like time manipulation, area manipulation, land-sea masking, variable derivation, ... .
+{: .callout}
+
+> ## ESMValTool can’t locate the data
+>
+> You are assisting a researcher with ``ESMValTool``. The researcher changes the ``project`` entry to ``CMIP6``
+and runs the recipe. However, ESMValTool encounters an error like:
+>
+~~~bash
+ esmvalcore._recipe_checks.RecipeError: Missing data
+2020-06-29 17:26:41,303 UTC [43830] INFO    If you suspect this is a bug or need help,
+please open an issue on https://github.com/ESMValGroup/ESMValTool/issues and
+attach the run/recipe_*.yml and run/main_log_debug.txt files from the output directory.
+~~~
+What suggestions would you give the researcher for fixing the error?
+>
+>> ## Solution
+>>
+>> 1. inspect the ``main_log.txt``
+>> 2. check the user-config.yml to see if the correct directory for input data is introduced.
+>> 3. check the available data, regarding exp, mip, ensemble, start_year, and end_year
+>> 4. check the variable name in the ``diag_timeseries_temperature`` section in the recipe.
+> {: .solution}
+{: .challenge}
+
+> ## Check pre-processed data
+>
+> The setting ``save_intermediary_cubes`` in the configuration file can be used
+save the pre-processed data. More information about this setting can be found at
+[Configuration]({{ page.root }}{% link _episodes/03-configuration.md %})
+{: .callout}
+
+The result of the pre-processors is passed to the ``diagnostic_timeseries.py`` script,
```

Comment:
```suggestion
The result of the pre-processor is passed to the ``diagnostic_timeseries.py`` script,
```
I think the general terminalogy used in the ESMValTool community is "the preprocessor" (singular), which consists of a number of "preprocessor steps" or "preprocessor functions"